🌱 Bump CAPI to v1.10.5

mboersma opened the pull request:
**What this PR does / why we need it**:

Bumps the Cluster API dependency to release [v1.10.5](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.10.5).

**Which issue(s) this PR fixes**:

Fixes #371

mboersma commented:
/test 
pull-cluster-api-addon-provider-helm-test-main

mboersma commented:
/test pull-cluster-api-addon-provider-helm-test-main

mboersma commented:
/test pull-cluster-api-addon-provider-helm-verify-main
/test pull-cluster-api-addon-provider-helm-build-main
/test pull-cluster-api-addon-provider-helm-test-mink8s-main

mboersma commented:
/test pull-cluster-api-addon-provider-helm-test-main
/test pull-cluster-api-addon-provider-helm-test-mink8s-main
/test pull-cluster-api-addon-provider-helm-build-main

mboersma commented:
/test pull-cluster-api-addon-provider-helm-verify-main
/test pull-cluster-api-addon-provider-helm-test-main
/test pull-cluster-api-addon-provider-helm-test-mink8s-main
/test pull-cluster-api-addon-provider-helm-build-main

dmvolod commented on api/v1alpha1/helmchartproxy_webhook.go:
```diff
@@ -37,19 +37,19 @@ var helmchartproxylog = logf.Log.WithName("helmchartproxy-resource")
 func (r *HelmChartProxy) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
+		WithDefaulter(r).
```
@mboersma What do you think about moving web-hooks to the separate package and exclude it from the apidiff calculation to avoid confusion?

Jont828 commented:
Let me know if you need help debugging the e2e test failures!

mboersma commented:
/retitle 🌱 Bump CAPI to v1.10.1

Jont828 commented:
@mboersma Are you still working on this?

mboersma commented:
@Jont828 thanks for the nudge, I updated this PR but there are still some failures. Let me know if you have any ideas. I think I've updated all the metadata files correctly but it seems like that kind of failure.

dmvolod commented on go.mod:
```diff
@@ -144,66 +142,64 @@ require (
 )
 
-replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.9.6
+replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.10.0
```
It seems to `get_go_version` function in Makefile doesn't support module replacement.

Do we really need to override Cluster API to the version v1.10.0 not v1.10.3?

mboersma replied:
Oops, no--this was intended to be a global search-and-replace but it appears I failed to update this one. Thanks for reviewing! I updated it to v1.10.3.

mboersma commented on config/crd/kustomization.yaml:
```diff
@@ -6,26 +6,28 @@ resources:
 #+kubebuilder:scaffold:crdkustomizeresource
 
-patchesStrategicMerge:
 # [WEBHOOK] To enable webhook, uncomment all the sections with [WEBHOOK] prefix.
```
I'm going to take another pass at the `kustomization.yaml` files and see if some comments should be removed (or preserved), because this regeneration made them less organized (although still valid).

mboersma commented:
@Jont828 @dmvolod @hrak @nojnhuh I think this is finally ready for review. 😅 Thanks for the assist @hrak!

I agree we could move webhooks to a separate package to keep apidiff from complaining, but if we're ok with deferring that this can go in a minor release.
